src/core/server/http/base_path_service.ts

```diff
@@ -22,13 +22,15 @@ import { modifyUrl } from '../../utils';
 
 export class BasePath {
   private readonly basePathCache = new WeakMap<LegacyRequest, string>();
+  public readonly serverBasePath: string;
```

Comment:
:tada: (we may need to explicitly define it in `interface CoreSetup` as well)

Reply:
Where ever the public-facing type for this is, can we make sure there's a TSDoc comment explaining what this is and why you would need to use it?